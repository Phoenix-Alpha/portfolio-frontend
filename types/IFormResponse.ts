export interface IFormResponse {
  isSuccess: boolean
  message?: string
}
