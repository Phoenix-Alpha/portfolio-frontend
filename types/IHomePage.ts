export interface IHomePage {
  heroContent?: string
}
