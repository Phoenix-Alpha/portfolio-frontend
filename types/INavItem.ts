export interface INavItem {
  key: string
  title: string
  url?: string
  isGroup?: boolean
  divider?: boolean
  subItems?: INavItem[]
  external?: string
  target?: string
}
