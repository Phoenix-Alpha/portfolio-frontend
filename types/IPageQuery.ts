import { DocumentNode } from 'graphql'
import { AnyAction } from 'redux'

export interface IPageQuery {
  query: DocumentNode
  variables?: Record<string, unknown>
  context?: Record<string, unknown>
  path?: string
  normalizer?: (pathEntity?: unknown) => unknown
  action?: (pathEntity?: unknown) => AnyAction
  actions?: Array<(pathEntity?: unknown) => AnyAction>
}
