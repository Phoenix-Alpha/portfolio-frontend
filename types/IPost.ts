export interface IPost {
  id: string
  title: string
  description: string
  slug: string
  date?: string
  cover?: string
  tags?: string[]
  readTimeMins?: number
  content?: string
}
