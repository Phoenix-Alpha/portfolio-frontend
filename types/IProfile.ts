export interface IProfile {
  firstName: string
  lastName: string
  email: string
  headline?: string
  bio?: string
  avatar?: string
  country?: string
  state?: string
  city?: string
  address?: string
}
