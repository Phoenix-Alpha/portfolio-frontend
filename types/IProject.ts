export interface IProject {
  id: string
  title: string
  description: string
  slug: string
  link?: string
  cover?: string
  tags?: string[]
  content?: string
}
