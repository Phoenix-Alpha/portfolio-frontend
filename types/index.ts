export * from './pages'
export * from './IRawUploadFile'
export * from './IPageQuery'
export * from './IProfile'
export * from './IStoreState'
export * from './TTheme'
export * from './INavItem'
export * from './IFormResponse'
