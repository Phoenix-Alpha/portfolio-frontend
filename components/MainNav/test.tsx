import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { INavItem } from '@/types'

import { MainNav } from './index'

describe('<MainNav />', () => {
  it('should renders correct hidden <MainNav />', () => {
    const isShown = false
    const setIsShown = jest.fn()
    const activeItem: INavItem = {
      key: 'test-nav-item-key',
      title: 'Test Item'
    }
    const toggleMainNavItem = jest.fn()

    const wrapper = shallow(
      <MainNav
        isShown={isShown}
        setIsShown={setIsShown}
        activeItem={activeItem}
        toggleMainNavItem={toggleMainNavItem}
      />
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
