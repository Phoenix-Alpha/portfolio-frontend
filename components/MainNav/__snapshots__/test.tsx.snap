// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNav /> should renders correct hidden <MainNav /> 1`] = `
<nav
  className="motion-reduce:animate-none animate-slide-in-left fixed inset-0 pb-16 lg:animate-none lg:flex lg:flex-grow lg:items-center lg:pt-0 lg:py-0 lg:static lg:ml-16 bg-[#0A1134] lg:bg-transparent hidden"
>
  <CloseButton
    className="lg:hidden absolute top-6 right-5 w-3 h-3"
    colorHex="#fff"
    height={15}
    onClick={[Function]}
    width={15}
  />
  <div
    className="lg:hidden py-[18px] px-5 border-b border-[#262C50]"
  >
    <SiteLogo />
  </div>
  <div
    className="lg:mr-auto lg:mb-0"
  >
    <ul
      className="flex flex-col lg:flex-row lg:mr-auto"
    >
      <MainNavItem
        active={false}
        handleClick={[MockFunction]}
        item={
          Object {
            "key": "main-nav-key-about",
            "title": "About",
            "url": "/about-us",
          }
        }
        key="main-nav-key-about"
      />
      <MainNavItem
        active={false}
        handleClick={[MockFunction]}
        item={
          Object {
            "key": "main-nav-key-faq",
            "title": "FAQ",
            "url": "/faq",
          }
        }
        key="main-nav-key-faq"
      />
      <MainNavItem
        active={false}
        handleClick={[MockFunction]}
        item={
          Object {
            "key": "main-nav-key-contact",
            "title": "Contact",
            "url": "/contact",
          }
        }
        key="main-nav-key-contact"
      />
    </ul>
  </div>
  <Divider
    className="lg:hidden mx-4 bg-[#262C50]"
  />
  <div
    className="lg:flex lg:items-center lg:ml-7"
  >
    <ul
      className="lg:flex lg:items-center"
    >
      <MainNavItem
        active={false}
        handleClick={[MockFunction]}
        item={
          Object {
            "key": "main-nav-key-raise-capital",
            "title": "Raise Capital",
            "url": "/raise-capital",
          }
        }
        smallOnMobile={true}
      >
        <RaiseCapitalIcon
          className="inline mr-2"
        />
      </MainNavItem>
    </ul>
    <Divider
      className="lg:hidden mx-4 bg-[#262C50]"
    />
    <Divider
      className="hidden lg:inline-block h-[1rem] bg-[#484F79]"
      isVertical={true}
    />
    <ul
      className="lg:flex lg:items-center"
    >
      <MainNavItem
        active={false}
        handleClick={[MockFunction]}
        item={
          Object {
            "key": "main-nav-key-sign-in",
            "title": "Sign In",
            "url": "/auth/signin",
          }
        }
        smallOnMobile={true}
      />
    </ul>
    <div
      className="px-4 lg:px-0 lg:ml-4 w-full lg:w-[7.125rem]"
    >
      <ButtonLink
        className="py-4 w-full h-full text-sm lg:font-semibold rounded-sm lg:rounded-none"
        href="/auth/signup"
      >
        Sign Up
      </ButtonLink>
    </div>
  </div>
</nav>
`;
